exe/Main.hs

```diff
@@ -224,14 +240,39 @@ getComponentOptions cradle = do
 
 
 createSession :: ComponentOptions -> IO HscEnvEq
-createSession opts = do
+createSession (ComponentOptions theOpts _) = do
     libdir <- getLibdir
+
+    cacheDir <- Main.getCacheDir theOpts
+
     env <- runGhc (Just libdir) $ do
-        _targets <- initSession opts
+        dflags <- getSessionDynFlags
+        (dflags', _targets) <- addCmdOpts theOpts dflags
+        _ <- setSessionDynFlags $
+             -- disabled, generated directly by ghcide instead
+             flip gopt_unset Opt_WriteInterface $
+             -- disabled, generated directly by ghcide instead
+             -- also, it can confuse the interface stale check
+             dontWriteHieFiles $
+             setHiDir cacheDir $
+             setDefaultHieDir cacheDir $
+             setIgnoreInterfacePragmas $
+             disableOptimisation dflags'
         getSession
     initDynLinker env
     newHscEnvEq env
 
+setIgnoreInterfacePragmas :: DynFlags -> DynFlags
+setIgnoreInterfacePragmas df =
+    gopt_set (gopt_set df Opt_IgnoreInterfacePragmas) Opt_IgnoreOptimChanges
```

Comment:
Why do we need `IgnoreOptimChanges` if we set the opt level to 0?

Reply:
We don't _need_ to. I can remove it if needed